src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java

```diff
@@ -205,6 +211,20 @@ public void connectBaseParticipationsToExerciseAndSave(ProgrammingExercise progr
         programmingExercise.setSolutionParticipation(solutionParticipation);
     }
 
+    private void connectAuxiliaryRepositoriesToExerciseAndSave(ProgrammingExercise exercise) {
+        List<AuxiliaryRepository> savedRepositories = new ArrayList<>(exercise.getAuxiliaryRepositories().stream()
+            .filter(repo -> repo.getId() != null).toList());
+        exercise.getAuxiliaryRepositories().stream()
+            .filter(repository -> repository.getId() == null)
+            .forEach(repository -> {
+                repository.setExercise(null);
+                repository = auxiliaryRepositoryRepository.save(repository);
+                repository.setExercise(exercise);
```

Comment:
This is strange, especially because the method is called `connectAuxiliaryRepositoriesToExerciseAndSave`. I would at least add a comment about that here if this is required (and why it works).